fridge-job-api/Dockerfile

```diff
@@ -1,6 +1,23 @@
 FROM ghcr.io/astral-sh/uv:python3.13-bookworm
 
-COPY . /app
+RUN groupadd -g 1001 appuser && useradd -m -u 1001 -g appuser appuser
+
 WORKDIR /app
-RUN uv sync --locked
-CMD ["uv", "run", "fastapi", "run", "app/main.py"]
+# Install dependencies
+RUN --mount=type=cache,target=/root/.cache/uv \
+    --mount=type=bind,source=uv.lock,target=uv.lock \
+    --mount=type=bind,source=pyproject.toml,target=pyproject.toml \
+    uv sync --locked --no-install-project --no-editable
+ADD . /app
+
+# Sync the project
+RUN --mount=type=cache,target=/root/.cache/uv \
+    uv sync --locked --no-editable
+
+# Place executables in the environment at the front of the path
+ENV PATH="/app/.venv/bin:$PATH"
+ENTRYPOINT []
+
+RUN chown -R appuser:appuser /app
```

Comment:
This seems strange. Why do this over copying the app and installing the dependencies?

Reply:
Ok, seems fine to follow that convention if there is a benefit.